### PR TITLE
[NoJira] Cards are no buttons anymore

### PR DESCRIPTION
**Pull request by frugoman**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack-SwiftUI/Card/Classes/BPKCard.swift [hunk 1]
```diff
@@ -36,6 +36,8 @@ public struct BPKCard<Content: View>: View {
     private let padding: Padding
     private let cornerStyle: CornerStyle
     private var tapAction : () -> Void = {}
+    
+    @State private var selectionOverlayOpacity = 0.0
```
Comment (frugoman) [inline Backpack-SwiftUI/Card/Classes/BPKCard.swift:40]: Added this as an effect to be had when the user clicks on a card, although it's not the exact same feeling as when it was a button :(

(we cannot support highlighting if it's not a button, that's why we went with buttons back then!)

File: Backpack-SwiftUI/Card/Classes/BPKCard.swift [hunk 2]
```diff
@@ -54,41 +56,31 @@ public struct BPKCard<Content: View>: View {
...
                     .animation(.easeInOut)
             )
             .clipShape(RoundedRectangle(cornerRadius: cornerRadius))
-            .animation(.easeOut(duration: 0.2), value: configuration.isPressed)
+            .shadow(elevation.shadow)
+            .onTapGesture {
+                selectionOverlayOpacity = 1
+                withAnimation {
+                    selectionOverlayOpacity = 0
+                }
```
Comment (frugoman) [inline Backpack-SwiftUI/Card/Classes/BPKCard.swift:74]: When the user taps, we animate opacity to simulate this

**Review by gert-janvercauteren — COMMENTED** [review]
From an a11y point of view, should we combine everything inside and add a button trait? 
Or allow the option to group or not, with a button trait when grouped?

File: Backpack-SwiftUI/Card/Classes/BPKCard.swift [hunk 3]
```diff
@@ -54,41 +56,33 @@ public struct BPKCard<Content: View>: View {
...
+            .shadow(elevation.shadow)
+            .onTapGesture {
+                selectionOverlayOpacity = 0.2
+                withAnimation {
+                    selectionOverlayOpacity = 0
+                }
+                tapAction()
+            }
+            .accessibilityElement(children: .combine)
+            .accessibilityAddTraits(.isButton)
```
Comment (gert-janvercauteren) [inline Backpack-SwiftUI/Card/Classes/BPKCard.swift:78]: Awesome!

**Review by gert-janvercauteren — APPROVED** [review]
LGTM

**Pull request merged**